### DISCUSSION TITLE
✨ Support 16kb page size for Android 15+ devices

### DISCUSSION THREAD

**Discussion by k-saparia** [software-mansion/react-native-reanimated#6362] [Ideas]
### Description
I am encountering issues when running my app on Android 15 devices with a 16 KB page size, particularly with some native components. Are there any plans to recompile the library to address these issues and ensure compatibility with phones that have a 16 KB page size?

As stated on Android [doc](https://developer.android.com/guide/practices/page-sizes#compile-16-kb-alignment),

> Beginning with Android 15, Android supports devices that are configured to use a page size of 16 KB (16 KB devices). If your app uses any [NDK](https://developer.android.com/ndk) libraries, either directly or indirectly through an SDK, then you will need to rebuild your app for it to work on these 16 KB devices.

An analysis of my app using the APK Analyzer [as described by Android](https://developer.android.com/guide/practices/page-sizes#16-kb-impact) shows that it does use **libreanimated.so** as native code.
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/ec6e4319-7cd2-447a-b69b-a14902296293">
This is the native binary reported by the APK analysis:

libreanimated.so

### Describe the solution you'd like
I would like for **libReanimated** to support 16kb page size phones so that apps supporting sdk 35 do not crash
### Additional context
Android doc: https://developer.android.com/guide/practices/page-sizes#compile-16-kb-alignment
react-native thread: https://github.com/facebook/react-native/issues/45054
### What Platforms whould this feature/enhancement affect?
Android

### Alternatives/Workarounds
No workaround I have investigated so far.

**Comment by parthshiroya6** [comment]
**If you upgrade to Android 36 (targetSdkVersion 36) and encounter errors like:**

<img width="798" height="429" alt="Image" src="https://github.com/user-attachments/assets/be0374c3-b951-456d-ae81-86bafb9af3da" />

then add the following configuration in your ```android/app/build.gradle ```file to exclude the problematic libraries:

```
android {
    ndkVersion rootProject.ext.ndkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion
    compileSdk rootProject.ext.compileSdkVersion

    packagingOptions {
        // Add these lines to handle 16KB page size issue
        jniLibs {
            useLegacyPackaging = true
        }

        // Exclude problematic libraries if needed
        exclude "lib/x86_64/libandroidlame.so"
        exclude "lib/x86_64/libucrop.so"
    }
}
```
...

**Comment by rohan-h-7edge** [comment]
If your React Native app gets rejected on Play Store for the 16KB page size issue, this guide helped me fix it: https://community.7edge.com/t/react-native-android-16-kb-page-size-fix-avoid-play-store-rejection/340?u=rohanhalesh
